### PR TITLE
add from_dict() wrappers to spl/token/instructions.py

### PR DESCRIPTION
**Pull request by azerv1**
related #344

### CODE REVIEW

File: src/spl/token/instructions.py [hunk 1]
```diff
@@ -25,6 +29,11 @@ class AuthorityType(IntEnum):
     CLOSE_ACCOUNT = 3
     """Authority to close a token account."""
 
+    @classmethod
+    def from_dict(cls, params: dict) -> "AuthorityType":
+        """AuthorityType dict wrapper."""
+        return cls(**params)
+
```
Comment (michaelhly) [inline src/spl/token/instructions.py:36]:
```suggestion
```
This one is not necessary as it is an Enum

**Comment by michaelhly** [comment]
Would you mind updating the corresponding unit tests here?

**Comment by kevinheavey** [comment]
What does this achieve? Can't the user already call `ApproveParams(**params)`?

**Comment by azerv1** [comment]
you are right. I falsely assumed that you were asking for something like python coding style convention. But regarding #344 ,  can you explain what more pythonic is?

**Pull request closed without merging**